htdocs/projet/class/task.class.php

```diff
@@ -267,7 +267,11 @@ function fetch($id, $ref='', $loadparentdata=0)
 				}
 
 				// Retreive all extrafield data
-			   	$this->fetch_optionals();
+				$this->fetch_optionals();
+
+				if(! empty($conf->global->PROJECT_ALLOW_COMMENT_ON_TASK)) {
```

Comment:
fetchComments may return a very high number of lines and in mosts cases, we don't need them.
I would prefer to see the addition of fetchComments after the fetch where we need it instead of loading a subtable content into a CRUD fetch.

Reply:
The same behavior is already implemented in project class when fetching a project
It allow to show how many comments the task have next to tab name

Reply:
fetch of subtable is done in fetch only when child is a very limited list in number of record (for example list of lines of an invoice) and that has no reason to increase with time.
For time spent for example in project, they are not loaded into the fetch but into a fetch function called when required (function loadTimeSpent)
It is same for tracked events into agenda, they are loaded only when required.
It should be same for fetch_comments (that a similar feature of agenda tracked events)

I discover that fetching comment into tasks is also done into the fetch. I missed it and i should not validate it. If you can change this to into task, it will be welcome...  ;-)